### PR TITLE
Rewrite translation map root-find

### PR DESCRIPTION
**Pull request by wthrowe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Domain/CoordinateMaps/TimeDependent/Translation.cpp
Comment (AlexCarpenter46) [inline src/Domain/CoordinateMaps/TimeDependent/Translation.cpp:252]: I haven't tested this being taken out, but does this still cover the cases where the root could be or is close to 0?

**Review by AlexCarpenter46 — COMMENTED** [review]
Looks good, didn't realize I was passing the wrong value for the function_at_upper_bound. If this covers the case of the root being 0 or close to, then LGTM

File: src/Domain/CoordinateMaps/TimeDependent/Translation.cpp
Comment (wthrowe) [reply src/Domain/CoordinateMaps/TimeDependent/Translation.cpp:252]: It's hard to analyze this code with so many user-specified functions (`f_of_r_` and multiple functions of time).  Can you give an example of the sort of case you're worried about?

I'm also not entirely sure what `root` represents.  It appears to be some sort of radius, but I don't know of what.

Comment (AlexCarpenter46) [reply src/Domain/CoordinateMaps/TimeDependent/Translation.cpp:252]: The `root` here represents the radius for the radial math function that will take the `target_coords` back to the `source_coords`. So if you have a point that gets translated in a way that looks very close to `gsl::at(result, i) += gsl::at(function_of_time_, i) * 1` (like the case when there's no radial falloff) then the root has to be 0. I think this can happen when the radius to your original coordinate is close to the origin and hasn't been translated very much yet.

Comment (wthrowe) [reply src/Domain/CoordinateMaps/TimeDependent/Translation.cpp:252]: I see.  Not sure if there might have been problems there, so I rewrote the root-find so that the function is manifestly non-positive at r=0.

File: src/Domain/CoordinateMaps/TimeDependent/Translation.cpp
Comment (AlexCarpenter46) [inline src/Domain/CoordinateMaps/TimeDependent/Translation.cpp]: square(gsl::at(distance_to_center, i) - f_of_r * gsl::at(function_of_time, i));

File: src/Domain/CoordinateMaps/TimeDependent/Translation.hpp
Comment (AlexCarpenter46) [inline src/Domain/CoordinateMaps/TimeDependent/Translation.hpp:65]: should this be `r^2 = \vec{\bar{\xi}}^2 - \Big(\vec{c} - F(r)\vec{T}(t)\Big)^2`

**Review by AlexCarpenter46 — APPROVED** [review]
Ran 10,000 tests without a fail, LGTM once gls::at is in 😄! Also thanks for doing this, the root function is much easier to follow now

File: src/Domain/CoordinateMaps/TimeDependent/Translation.cpp
Comment (wthrowe) [reply src/Domain/CoordinateMaps/TimeDependent/Translation.cpp]: Did the first, but not the second.  We generally don't use `gsl::at` on `DataVector`s because it's redundant.

File: src/Domain/CoordinateMaps/TimeDependent/Translation.cpp
Comment (knelli2) [inline src/Domain/CoordinateMaps/TimeDependent/Translation.cpp]: [Optional] No need for `distance_to_center` here. Just do

```
const double root = root_finder(target_coords - center_, function_of_time);
```

**Review by knelli2 — DISMISSED** [review]

File: src/Domain/CoordinateMaps/TimeDependent/Translation.cpp
Comment (wthrowe) [reply src/Domain/CoordinateMaps/TimeDependent/Translation.cpp]: Good point.  Will do.

**Review by knelli2 — APPROVED** [review]

**Pull request merged**
